### PR TITLE
Add create_system_build_webhook

### PR DESCRIPTION
**Pull request by nottrobin**
This provides a method, `create_system_build_webhook`, which will first check if a webhook for the existing URL and system build endpoint already exists, and if it doesn't create it.

This is so we can create webhooks on the fly in the ubuntu.com image-builder flow.

QA
--

Can be tested with this branch: https://github.com/canonical-web-and-design/ubuntu.com/pull/7245

### CODE REVIEW

File: canonicalwebteam/launchpad/models.py [hunk 1]
```diff
@@ -64,18 +68,70 @@ def __init__(self, username, token, secret, session):
...
+
+        system_year = re.match(r"^[^\d]+(?:64)?(\d{2})(\.\d{2})?$", system)[1]
+        codename = self.system_codenames[system_year]
+        project = "ubuntu-core"
+        if system.startswith("classic"):
+            project = "ubuntu-cpc"
+
+        webhooks = self.get_collection_entries(
+            "https://api.launchpad.net/devel/"
+            f"~{self.username.replace('.', '')}/"
```
Comment (jkfran) [inline canonicalwebteam/launchpad/models.py:105]: @nottrobin just curiosity, why the replace?

Comment (nottrobin) [reply canonicalwebteam/launchpad/models.py:105]: Launchpad seems to need it. No idea why.

Comment (jkfran) [reply canonicalwebteam/launchpad/models.py:105]: The username that we use in Snapcraft is `build.snapcraft.io` and I didn't have any issue with the dots and the username that you use is `imagebuild` without any dots, right?

https://launchpad.net/~imagebuild

Comment (nottrobin) [reply canonicalwebteam/launchpad/models.py:105]: I have no idea. But for some reason, the name here is `imagebuild` and yet when we log in to the API we have to use the form `image.build`. Maybe it's something about this user specifically then? In which case I suppose I'd need to add an extra parameter, and this conditional is wrong. Worth digging into. Not sure I have time right now.

Comment (jkfran) [reply canonicalwebteam/launchpad/models.py:105]: No worries, interesting, my concern is that we need our user with the dots in snapcraft but if the test is passing I guess you are doing the replace only for your endpoints.

**Pull request merged**